.github/workflows/deploy.yaml

```diff
@@ -535,6 +545,46 @@ jobs:
           slack-channel-id: ${{ inputs.slack-channel-id }}
           slack-message-id: ${{ steps.publish-slack.outputs.slack-message-id }}
 
+  deploy_by_argocd:
+    if: ${{ ! inputs.push-to-manifests }}
+    name: Deploy (ArgoCD)
+    needs: push-manifest-list
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - name: Publish progress message to slack
+        uses: monta-app/slack-notifier-cli-action@main
+        id: publish-slack
+        with:
+          job-type: "deploy"
+          job-status: "progress"
+          service-name: ${{ inputs.service-name }}
+          service-emoji: ${{ inputs.service-emoji }}
+          slack-app-token: ${{ secrets.SLACK_APP_TOKEN }}
+          slack-channel-id: ${{ inputs.slack-channel-id }}
+          slack-message-id: ${{ needs.push-manifest-list.outputs.slack-message-id }}
+      - name: Download ArgoCD CLI
+        shell: bash
+        run: |
+          curl -sSL -o /argocd https://github.com/argoproj/argo-cd/releases/latest/download/argocd-linux-amd64
+          sudo chmod +x /argocd
+      - name: ArgoCD update application
+        shell: bash
+        run: |
+          OLD_TAG=$(/argocd --auth-token ${{ secrets.ARGOCD_TOKEN }} --server ${{ inputs.argocd-server }} app get argocd/wallet-dev --show-params | grep image\.tag | awk '{print $2}')
+          /argocd --auth-token ${{ secrets.ARGOCD_TOKEN }} --server ${{ inputs.argocd-server }} app patch ${{ inputs.service-identifier }}-${{ inputs.stage }} --patch="[{\"op\": \"replace\", \"path\": \"/spec/info/0\", \"value\": {\"name\": \"Source Code Diff\", \"value\": \"https://github.com/${{ github.repository }}/compare/${OLD_TAG::8}...${GITHUB_SHA::8}\"}}]"
```

Comment:
can't we do single-quotes here in the surrounding part, so that we don't need all this escaping?

Reply:
Hehe, beautiful 🤩 Guess all your key-smashing earlier today didn't pay off 😂

Reply:
Or it did, because without them I wouldn't learn of this now :)